### PR TITLE
Fixes #3349 - Updates documentation for webpack and JS unit tests

### LINKED ISSUE #3349: Update documentation to reflect webpack changes
Update documentation on dev environment setup:
https://github.com/webcompat/webcompat.com/blob/master/docs/dev-env-setup.md

Add js unit test section in 
https://github.com/webcompat/webcompat.com/blob/master/docs/tests.md

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #3349 

## Proposed PR background

* Adds documentation for JS  unit tests
* adds documentation for webpack

### CODE REVIEW

**Comment by karlcow** [comment]
@ksy36 I just noticed that the tests were not in the test directory? Was there a technical difficulties for not putting them in the tests directory?

**Comment by ksy36** [comment]
Ahh thanks for adding the docs :) For the tests, personally it's more convenient if the unit test is in the same location as the tested function, without switching context, so I decided to keep them together.

File: docs/dev-env-setup.md [hunk 1]
```diff
@@ -206,52 +206,23 @@ found 6 vulnerabilities (2 low, 4 high)
   run `npm audit fix` to fix them, or `npm audit` for details
 ```
 
-Then we will compile the assets with `grunt`.
+Then we will compile the assets with `webpack` by running `npm run prod`.
```
Comment (ksy36) [inline docs/dev-env-setup.md:209]: here we need to run `npm run build`, it'll run `webpack --config ./webpack/webpack.dev.js`
(`npm run prod` compiles the production build with minification and some optimizaitions for production)

Comment (ksy36) [reply docs/dev-env-setup.md:209]: Actually I've just noticed this part exists already:
https://github.com/webcompat/webcompat.com/blob/456791f3d6e4076f6f3f304349bbb887a3a29913/docs/dev-env-setup.md#installing-npm-dependencies
Maybe we can delete it then?

**Comment by ksy36** [comment]
We also need to update this section https://github.com/webcompat/webcompat.com/blob/456791f3d6e4076f6f3f304349bbb887a3a29913/docs/dev-env-setup.md#building-the-project
I'll do it, after this PR is merged

File: docs/dev-env-setup.md (hunk 1 above)
Comment (karlcow) [reply docs/dev-env-setup.md]: removed the assets section

**Comment by karlcow** [comment]
Thanks @ksy36

**Pull request merged**